File: android/src/main/java/org/maplibre/reactnative/components/layers/types/circle/MLRNCircleLayerManager.java
package org.maplibre.reactnative.components.layers.types.circle;

import com.facebook.react.uimanager.ThemedReactContext;

import org.maplibre.reactnative.components.layers.MLRNVectorLayerManager;

public class MLRNCircleLayerManager extends MLRNVectorLayerManager<MLRNCircleLayer> {
    public static final String REACT_CLASS = "MLRNCircleLayer";

    @Override
    public String getName() {
        return REACT_CLASS;
    }

    @Override
    protected MLRNCircleLayer createLayerViewInstance(ThemedReactContext reactContext) {
        return new MLRNCircleLayer(reactContext);
    }
}

File: android/src/main/java/org/maplibre/reactnative/components/layers/types/circle/MLRNCircleLayerManager.kt
package org.maplibre.reactnative.components.layers.types.circle

import com.facebook.react.uimanager.ThemedReactContext
import org.maplibre.reactnative.components.layers.MLRNVectorLayerManager

class MLRNCircleLayerManager : MLRNVectorLayerManager<MLRNCircleLayer?>() {
    override fun getName(): String {
        return REACT_CLASS
    }

    override fun createLayerViewInstance(reactContext: ThemedReactContext?): MLRNCircleLayer {
        return MLRNCircleLayer(reactContext)
    }

    companion object {
        const val REACT_CLASS: String = "MLRNCircleLayer"
    }
}
